### ISSUE TITLE
[dataio] add docs and unit tests for zstd (Trac #2011)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2003]
tests should include reading and writing. Docs should explain the different compression levels and include  plots justifying this algorithm 

Migrated from https://code.icecube.wisc.edu/ticket/2011
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "tests should include reading and writing. Docs should explain the different compression levels and include  plots justifying this algorithm ", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo core", 
    "summary": "[dataio] add docs and unit tests for zstd", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2017-05-09T18:48:18", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170509T18:49:59: cweaver changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20180613T18:23:45: kjmeagher changed milestone from "IceRec 5.2" to "Autumnal Equinox 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180922T10:17:21: 
* olivas changed status from "accepted" to "assigned"
* olivas changed owner from "cweaver" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20180922T10:17:32: olivas changed milestone from "Autumnal Equinox 2018" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20181114T14:42:37: olivas changed owner from "olivas" to ""

**Comment by icecube-trac** [comment]
Trac update at 20181205T17:56:13: jbraun changed owner from "" to "jbraun"

**Comment by icecube-trac** [comment]
Trac update at 20181205T19:00:18: 
* jbraun commented:

> In changeset 3116:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="3116"
> Update dataio docs for ZSTD.  I'm not including plots justifying ZSTD.  This information can be found elsewhere.  Closes #2003.
> ```


* jbraun changed status from "assigned" to "closed"
* jbraun changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T22:03:05: nwhitehorn commented:

> In changeset 169270:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169270"
> Update dataio docs for ZSTD.  I'm not including plots justifying ZSTD.  This information can be found elsewhere.  Closes #2003.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas commented:

> Milestone Winter Solstice 2018 deleted


* olivas changed milestone from "Winter Solstice 2018" to ""
